docker/docker_client.go

```diff
@@ -253,6 +255,13 @@ func newDockerClient(sys *types.SystemContext, registry, reference string) (*doc
 		}
 		if reg != nil {
 			skipVerify = reg.Insecure
+		} else {
+			// unless specified in the SystemContext or in the
+			// config, set skipVerify to true if the registry
+			// is running on the localhost
+			if strings.HasPrefix(registry, "localhost:") || strings.HasPrefix(registry, "127.0.0.1") {
```

Comment:
This does not match `localhost`, and does match `127.0.0.19`.

Reply:
I am not sure if `localhost` would actually work since `localhost/image:latest` would be seen as an unqualified image and be transformed to `docker.io/localhost/image:latest` - Docker behaves like this as well.

Reply:
1. No, the `docker.io[/library]` normalization special-cases the `localhost` host name (other strings must contain a dot to be treated as host names, `localhost` is accepted as a special case – search in `c/image/docker/reference`)
2. This is checking `registry`, not `reference`; the caller must already be setting `registry` to the right host:port (not host:port/ns/repo) in order for connections to work.

To clarify, the review comment was primarily pointing out the inconsistent treatment of `:` (i.e. both ports, and non-port suffixes) in the two cases of the condition.

Reply:
Coming back to this issue. @mtrmac, shall we support the entire range from 127.0.0.1 through 127.255.255.255?

> This does not match localhost, and does match 127.0.0.19.

^ made me wonder.

Reply:
I’d rather not; all of these “don’t be secure” defaults are risky.

We get (guessing with no evidence) 98% of the convenience benefit matching just `localhost`, 1% matching `127.0.0.1`, with hopefully manageable risk.

Anyone in a special enough situation that there are different servers running on `127.0.0.1`, `127.0.0.19` and `127.255.255.254` has been doing enough manual and unusual configuration that having to explicitly configure `127.0.0.19` as insecure is negligible extra burden, and that we don’t quite know what/why is going on and whether the risk is still acceptable.